viewer/src/data/repository/RxModelSectorsRepository.ts

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+
+import { Observable, OperatorFunction, pipe, partition, merge, from } from 'rxjs';
+import { publish, flatMap, mergeAll, shareReplay, take, map } from 'rxjs/operators';
+import { ModelSectorService } from '../network/client/ModelSectorService';
+import { ModelSectorFormat } from '../parser/ModelSectorData';
+import { RxModelSectorParser } from '../parser/RxModelDataParser';
+
+interface ModelSector {
+  modelId: number;
+  file: string;
+  format: ModelSectorFormat;
+  toString(): string;
+}
+
+export class RxModelSectorsRepository<T> {
+  private readonly modelService: ModelSectorService;
+  private readonly sectorDataParser: RxModelSectorParser<T>;
+  private readonly sectorCache: { [modelSector: string]: Observable<T> } = {};
+
+  constructor(modelService: ModelSectorService, sectorDataParser: RxModelSectorParser<T>) {
+    this.modelService = modelService;
+    this.sectorDataParser = sectorDataParser;
+  }
+
+  getSectors(): OperatorFunction<ModelSector[], any> {
+    return pipe(
+      mergeAll(20),
```

Comment:
Is there a rate limit in place anywhere? If not, what is the effect of mergeAll?

Reply:
mergeAll() is a limit, but limit is weird in rxjs.

If I say a limit 2, then lower down limit 2, the first one processes two, and those two process two of their own, so 4 concurrent operations are in flight.

Reply:
Ah, that's right. Thanks :)